Update community-plugins.json

banisterious opened the pull request:
Adding custom-selected-word-count

# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/banisterious/obsidian-custom-selected-word-count

## Release Checklist
- [ ] I have tested the plugin on
  - [X]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [X] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [X] `main.js`
  - [X] `manifest.json`
  - [X] `styles.css` _(optional)_
- [X] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [X] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [X] My README.md describes the plugin's purpose and provides clear usage instructions.
- [X] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [X] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [X] I have added a license in the LICENSE file.
- [X] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

banisterious commented:
A CHANGELOG.md file has been added to the repository, documenting the initial release (v1.0.0) and all included features. This should provide clear version history and help users and reviewers track future changes. Please let me know if there are any further requirements or suggestions!

banisterious commented:
I've added comprehensive user documentation in [`USAGE.md`](docs/USAGE.md) since submitting this PR. This includes:
- Detailed usage instructions with annotated screenshots
- Cross-references from [`README.md`](README.md) and [`SPECIFICATION.md`](docs/SPECIFICATION.md)
- Clear examples of the plugin's functionality
- FAQ section for common scenarios

These additions should make the plugin more user-friendly and easier to understand. Please let me know if you'd like me to make any adjustments to the documentation.

banisterious commented:
Hello! I've implemented all the required changes from the review:

1. **Class Renaming**
- Renamed generic classes to more descriptive names (e.g., `CustomSelectedWordCountPlugin`)
- Updated all related references throughout the codebase

2. **Inline Styles**
- Moved all inline styles to dedicated CSS file
- Organized styles into logical sections with clear comments
- Added proper CSS class naming for better maintainability
- Improved theme compatibility using CSS variables
- Added responsive design for mobile devices

4. **Console Logging**
- Implemented proper debug logging system with settings toggle
- Replaced direct `console.log` calls with `debugLog` function
- Added structured logging with plugin context

All changes have been documented in the [CHANGELOG.md](./CHANGELOG.md) under the initial v1.0.0 release. The build date has also been updated to reflect today's date (May 7, 2025).

Please let me know if any further adjustments are needed!

joethei commented:
["dependencies": {](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/manifest.json#L11-L13), ["dependencies": {](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/manifest.json#L11-L13)
These are not valid parts of the manifest, remove them.

[document.body.addClass('custom-word-count-plugin');](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L315)
Remove this useless code.

[// Set up ribbon button if enabled](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L328-L331), [if (this.ribbonButton) {](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L532-L534), [// Ribbon Button Setting](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L772-L783)
As of v1.1.0 of Obsidian, the ribbon can be customized within Obsidian. So users can choose what items to include and in what order all in one place. Because of this, we're recommending that new plugins avoid introducing their own toggles for ribbon items.

['Count Selected Words',](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L420), [text: 'Clear History'](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L674), [containerEl.createEl('h3', { text: 'UI Elements' });](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L770),[.setName('Show Count in Status Bar')](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L788), [.setName('Enable Live Updates')](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L806), [.setName('Hide Core Word Count')](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L819), [.setName('Status Bar Label')](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L832), [containerEl.createEl('h3', { text: 'Path Exclusion' });](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L844), [.setName('Exclude Paths from Word Count')](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L847), [.setName('Exclude Windows Paths')](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L862), [text: '⚠️ Advanced: Custom Word Detection Regex (Expert Only)',](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L875), [.setName('Enable Advanced Regex (Expert Only)')](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L884), [text: 'Reset Test',](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L915)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[containerEl.createEl('h2', { text: 'Custom Selected Word Count Settings' });](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L767)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[containerEl.createEl('h3', { text: 'UI Elements' });](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L770),  [containerEl.createEl('h3', { text: 'Path Exclusion' });](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/85c0c41497cdb57a0a6c3823a675ba19facb3872/main.ts#L844)
For section headings in settings use:
```ts
new Setting(containerEl).setName('name here').setHeading();
```

banisterious commented:
Hello! I have made the requested changes and tested to make sure that all is still working as intended. I've also updated user and developer docs.

### Changes
- **Removed body class injections** - Eliminated all `document.body.addClass/removeClass` calls to comply with Obsidian guidelines
- **Updated settings interface** - Switched to `Setting.setHeading()` API and converted text to sentence case
- **Removed ribbon button and related setting** - Following Obsidian's requirement that ribbon buttons be an opt in feature
- **Fixed status bar visibility** - Corrected CSS rules that were hiding the plugin's status bar
- **Added missing settings** - Unix/UNC path exclusion, history options, and debug logging briefly dropped out during these changes; now restored
- **Cleaned up dependencies** - Removed `style-settings` from manifest
- **Reorganized documentation** - New structure with kebab-case naming and updated README

### Technical notes
- All functionality preserved - no breaking changes for users
- CSS updated with proper `word-count-` prefixing to prevent conflicts
- Replaced body class dependencies with direct element manipulation

Please let me know if I missed anything and/or inadvertently introduced more issues. Thanks!

banisterious commented:
I have added the individual files.

joethei commented:
["main": "main.js"](https://github.com/banisterious/obsidian-custom-selected-word-count/blob/73b10022c713858ba715b8d41c694257afd0e321/manifest.json#L10C2-L10C19)
This is also not a valid part of the manifest.
